algorithms/number_theory/eratosthenes_sieve.py

```diff
@@ -0,0 +1,59 @@
+"""Module with the implementation of eratosthenes sieve."""
+
+
+from typing import List
+
+
+class EratosthenesSieve:
+    """Class implementation of the eratosthenes sieve algorithm."""
+
+    def __init__(self, n: int):
+        self._n = n
+        self._no_cousins = [False for _ in range(self._n + 1)]
+
+    def eratosthenes_sieve(self) -> List[bool]:
+        """
+        Eratosthenes sieve algorithm.
+        """
+        self._no_cousins[0] = self._no_cousins[1] = True
+
+        for i in range(2, self._n + 1):
+            if not self._no_cousins[i]:
+                for j in range(i, self._n + 1):
+                    if i * j >= self._n + 1:
+                        break
+                    self._no_cousins[i * j] = True
+
+        return self._no_cousins
+
+    def _accum_cousins(self) -> List[int]:
+        """
+        Applies acumulative sum according to the no_cousins array.
+        """
+        dp = [0 for _ in range(self._n + 1)]
+        for i in range(1, self._n + 1):
+            dp[i] = dp[i - 1]
+            if not self._no_cousins[i]:
+                dp[i] += 1
+        return dp
+
+    def get_count_cousins(self, left: int, right: int) -> int:
+        """
+        Returns the count of prime numbers in the interval [left, right].
+
+        Parameters
+        ----------
+        left : int
+            Lower limit of range.
+        right : int
+            Upper limit of range.
+        """
+        count_cousins = 0
+        dp = self._accum_cousins()
```

Comment:
Consider doing this process just once and not every time you call the `get_count` method. I assume that this was to speed up the process of finding the number of primes in a range. But if we calculate the cumulative sum list every time we end up with linear complexity either way.

Let me know if you have any questions regarding this.